src/device/wg_controller.rs

```diff
@@ -382,10 +387,16 @@ async fn build_requested_peers_list<
             .filter(|network| features.ipv6 || network.is_ipv4())
             .collect();
 
+        let preshared_key = requested_state
+            .postquantum_wg
+            .as_ref()
+            .map(|pq| pq.pq_shared);
+
         if let Some(meshnet_peer) = requested_peers.get_mut(&exit_node.public_key) {
             // Exit node is meshnet peer, so just promote already existing node to be exit node
             // with allowed ips change
             meshnet_peer.peer.allowed_ips = allowed_ips;
+            meshnet_peer.peer.preshared_key = preshared_key;
```

Comment:
This does not really make sense for meshnet peer, as PQ and meshnet is not compatible. But it will not hurt also, lets just remember to remove it in the rekey PR ;).

Reply:
Yeah, let me do it in the followup

Reply:
Already done: https://github.com/NordSecurity/libtelio/pull/259/commits/cb5a90c22e310d821be689face0cc61dc189a808